invenio_search/api.py

```diff
@@ -83,7 +83,11 @@ class Meta:
 
     def __init__(self, **kwargs):
         """Use Meta to set kwargs defaults."""
-        kwargs.setdefault('index', getattr(self.Meta, 'index', None))
+        kwargs.setdefault(
```

Comment:
There are currently two ways to perform queries to different indices using this class:

1. Sub-classing and overriding `Meta.index`
2. Passing `index` explicitly as a keyword argument

The first case is of course covered... Though the usage of `kwargs.setdefault` instead of setting the value directly is probably because of the fact that this constructor was written before the need for prefixing index names "magically" existed... 

Maybe it would be better to make `Meta.index` some kind of property which relies on some `_index` field? @lnielsen WDYT?

Reply:
Locally I solved it by checking if the `kwargs['index']` is prefixed, and if not then adding it (at some point there's also a list passed to it instead a string...)